samcli/lib/build/workflow_config.py

```diff
@@ -117,6 +117,32 @@ def get_workflow_config(runtime, code_dir, project_dir):
                                           .format(runtime, str(ex)))
 
 
+def supports_build_in_container(config):
+    """
+    Given a workflow config, this method provides a boolean on whether the workflow can run within a container or not.
+
+    Parameters
+    ----------
+    config namedtuple(Capability)
+        Config specifying the particular build workflow
+
+    Returns
+    -------
+    tuple(bool, str)
+        True, if this workflow can be built inside a container. False, along with a reason message if it cannot be.
+    """
+
+    unsupported = {
+        DOTNET_CLIPACKAGE_CONFIG: "We do not support building dotnet Lambda functions within a container. Most "
```

Comment:
Should this be based on the work workflow or the runtime/lang? If tomorrow we add a second DotNet workflow, do we run the risk of not updating this too?

I don't know the right answer here. My initial thought is we might want to do this on a lang (all DotNet versions and workflows), to keep things consistent but could see a reason to constrain to the Workflow itself.

Reply:
Yeah I struggled with this as well. Ultimately, I realized that the workflow implementation is what dictates whether it can run within a container or not. So making the distinction based on workflow config seemed right here. We might find other cases to abstract it differently. Since this is an implementation detail, we can refactor later if needed